### PR TITLE
CommonComReader: Commonizing COM modules

### PR DESCRIPTION
**Pull request by thopiekar**
Now we can use PyWin32Com in case it is installed.
Downside: We can't choose between different SolidWorks versions. That's why I have to uninstall other versions I had installed for testing.

### CODE REVIEW

**Comment by nallath** [comment]
CURA-4267

File: CommonComReader.py [hunk 1]
```diff
@@ -16,14 +16,52 @@
...
+# Trying to import one of the COM modules
+try:
 
+    import win32com.client
+    import win32com.client.gencache
+    import pythoncom
+    import pywintypes
+
+    class ComFactory:
+        def CreateClassObject(app_name):
```
Comment (nallath) [inline CommonComReader.py]: codestyle; should be createClassObject

File: CommonComReader.py [hunk 2]
```diff
@@ -16,14 +16,52 @@
...
+            win32com.client.gencache.Rebuild()
+            #win32com.client.gencache.EnsureModule(app_name)
```
Comment (nallath) [inline CommonComReader.py]: Unused code.

File: CommonComReader.py [hunk 3]
```diff
@@ -16,14 +16,52 @@
...
+            return win32com.client.gencache.EnsureDispatch(app_name)
+            res = win32com.client.Dispatch("{F16137AD-8EE8-4D2A-8CAC-DFF5D1F67522}")
+            Logger.log("d", "DIR> %s", dir(res()))
```
Comment (nallath) [inline CommonComReader.py]: A bit clearer logging would be nice.

File: CommonComReader.py [hunk 4]
```diff
@@ -16,14 +16,52 @@
...
+except ImportError:
+    ## ComTypes
+    import comtypes
+    import comtypes.client
+
+    class ComFactory:
+        def CreateClassObject(app_name):
+            return comtypes.client.GetClassObject(app_name).CreateInstance()
+
+        def Coinit():
```
Comment (nallath) [inline CommonComReader.py]: Shouldn't this be coInitialize? (eg; co & init are different words right?)

Comment (thopiekar) [reply CommonComReader.py]: Good point. Will fix that..

**Comment by thopiekar** [comment]
Fixed and merged at: https://github.com/thopiekar/CuraDassaultSystemesPlugins/pull/12

**Pull request closed without merging**